sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -882,19 +883,31 @@ struct _pi_queue : _pi_object {
     uint32_t NextIndex{0};
   };
 
+  // When using standard commandlists, a group containing compute queue handles.
   pi_queue_group_t ComputeQueueGroup{this, queue_type::Compute};
 
-  // Vector of Level Zero copy command command queue handles.
-  // In this vector, main copy engine, if available, come first followed by
-  // link copy engines, if available.
+  // When using immediate command lists, a set of groups per thread id.
+  // Initialized with calling thread's id when queue is constructed.
+  std::map<std::thread::id, pi_queue_group_t> ComputeQueueGroupsByTID;
```

Comment:
can't the 2 be combined? if not using per-thread lists then there would be a single map entry

Reply:
It is useful to keep the single Compute and Copy queue group members as an initial value. They are set using env var settings for compute and copy engines at queue creation time. Later, as new threads start using the same queue, and new thread-specific Compute/Copy groups are created, the original Compute and Copy queue groups serve as an initializer for the per-thread ones. Otherwise, we would need to do the env var processing again for each new thread. Also, in non-per-thread mode we avoid the map lookup.